File: Python/Sorting/insertion-sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key
    return arr


# Testing
arr = [5, 4, 3, 2, 1]

print(insertion_sort(arr))  # [1, 2, 3, 4, 5]

File: JavaScript/Sorting/insertion-sort.js
function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let current = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > current) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = current;
    }
    return arr;
}

// Testing
let arr = [5, 4, 3, 2, 1];

console.log(insertionSort(arr)); // [1, 2, 3, 4, 5]
